### PR TITLE
remove CRD registration permissions from service accounts

### PR DESCRIPTION
**Pull request by mitchdraft**
addresses https://github.com/solo-io/squash/issues/155

going forward, user will have to register the DebugAttachment CRD with kubernetes prior to using squashctl in secure mode
This can be done by default (through user's own permissions) when using non-secure mode

### CODE REVIEW

**Comment by mitchdraft** [comment]
todo:
- default squashctl to the "with registration" client when not in secure mode
- add util for registering crd when using secure mode
- add warnings if clients have not yet been registered (secure mode)

File: cmd/logs/main.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package main
```
Comment (mitchdraft) [inline cmd/logs/main.go]: this file was for testing the skaffold log lib, will remove

File: pkg/config/squash.go [hunk 2]
```diff
@@ -383,6 +383,9 @@ func (s *Squash) debugPodFor() (*v1.Pod, error) {
 			HostPID:            true,
 			RestartPolicy:      v1.RestartPolicyNever,
 			NodeName:           targetPod.Spec.NodeName,
+			ImagePullSecrets: []v1.LocalObjectReference{{
```
Comment (mitchdraft) [inline pkg/config/squash.go]: added to support testing with a private image repo

File: pkg/install/squash.go [hunk 3]
```diff
@@ -97,29 +98,29 @@ func InstallSquash(cs *kubernetes.Clientset, namespace, containerRepo, container
 	privileged := true
 	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: SquashName,
+			Name: sqOpts.SquashPodName,
```
Comment (mitchdraft) [inline pkg/install/squash.go]: cleaned up this constant, refactored

**Comment by hgword** [comment]
👍

**Pull request merged**